File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02959/p02959_num6_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> monsters(N+1);
    for (int i = 0; i <= N; i++) {
        cin >> monsters[i];
    }

    vector<int> heroes(N);
    for (int i = 0; i < N; i++) {
        cin >> heroes[i];
    }

    int maxMonsters = 0;

    for (int i = 0; i < N; i++) {
        int defeatedMonsters = min(monsters[i], heroes[i]);
        maxMonsters += defeatedMonsters;
        monsters[i] -= defeatedMonsters;

        int extraDefeatedMonsters = min(monsters[i+1], heroes[i] - defeatedMonsters);
        maxMonsters += extraDefeatedMonsters;
        monsters[i+1] -= extraDefeatedMonsters;
    }

    cout << maxMonsters << endl;

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02959/p02959_num6_parsed.py
def max_monsters(N, A, B):
    max_monsters = 0

    for i in range(N):
        max_monsters += min(A[i], B[i])
        A[i] -= min(A[i], B[i])
        B[i] -= min(A[i], B[i])
        
        max_monsters += min(A[i+1], B[i])
        A[i+1] -= min(A[i+1], B[i])
        B[i] -= min(A[i+1], B[i])

    return max_monsters

# Read input
N = int(input())
A = list(map(int, input().split()))
B = list(map(int, input().split()))

# Calculate and print the maximum total number of monsters the heroes can defeat
print(max_monsters(N, A, B))